build.gradle.kts

```diff
@@ -38,6 +38,7 @@ dependencies {
   testImplementation("io.zonky.test:embedded-database-spring-test:2.0.1")
   testImplementation("io.zonky.test:embedded-postgres:1.3.0")
   testImplementation("io.projectreactor:reactor-test")
+  testImplementation("com.ninja-squad:springmockk:3.0.1")
```

Comment:
Do we really want another mocking framework? We have the Zonky stuff that would work for integration tests for DB too?

Reply:
It's not another mocking framework - it's a Spring extension for Mockk so that you can use the `@MockkBean` annotation in an integration test.

Reply:
The embedded database doesn't spare you from wiring in the repository. This means that in your test you would still have to write recalls to the database if you want to get the recalls. If you wire in a mock repository, like you do with `@MockkBean`, you don't have to write as much boiler plate to set up your data and also you can test some edge unhappy paths a bit easier, i.e. repository throws some weird exception.

Reply:
From my point of view I try to avoid using 'lateinit' as much as possible, always use a 'val' if possible, so anything that avoids it works for me :-) This style is always my preference:

`private val recallRepository = mockk<RecallRepository>()`

But I know some people like annotations, so not too fussed either way.

Reply:
I don't think you can use that code here. The `MockkBean` annotation also tells Spring that the RecallRepository bean in the context of this test should be instantiated as a mock.